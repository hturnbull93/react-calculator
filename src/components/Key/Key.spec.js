import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import Key from "./Key";

describe("Key", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Key keyAction={jest.fn()} keyType={""} keyValue={""} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a div", () => {
    const wrapper = shallow(
      <Key keyAction={jest.fn()} keyType={""} keyValue={""} />
    );
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("renders the keyValue", () => {
    const wrapper = shallow(
      <Key keyAction={jest.fn()} keyType={""} keyValue={""} />
    );
    wrapper.setProps({ keyValue: "test" });
    expect(wrapper.text()).toEqual("test");
  });
});
