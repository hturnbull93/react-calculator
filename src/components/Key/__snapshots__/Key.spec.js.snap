// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key renders correctly 1`] = `
<div
  className="key-container "
  onClick={[Function]}
>
  <p
    className="key-value"
  >
    
  </p>
</div>
`;
